### PR TITLE
Use LibYAML by default to increase performance in FileUtils

### PR DESCRIPTION
**Pull request by Higgcz**
I've noticed significant performance issues when using *file store* tracking backend compared to other frameworks. After profiling the code, I've noticed the `log_metric` method ([file_store.py:642](https://github.com/mlflow/mlflow/blob/master/mlflow/store/tracking/file_store.py#L642)) is loading YAML file with the run information to see if the run is currently active or not. This check is happening every time the `mlflow.log_metric` function is called.

By default, Python's YAML is using Python parser, which is significantly slower than its C++ counterpart. See [PyYAML Documentaion](https://pyyaml.org/wiki/PyYAMLDocumentation).

## What changes are proposed in this pull request?

I've added a try-except block into the `mlflow/utils/file_utils.py`, which tries to import the `CSafeLoader` and `CSafeDumper` to replace the Python's default ones. If this fails, it means *LibYAML* is not installed on the system and `file_utils.py` will import the default ones.

I also added small paragraph about *File store performance* under **Storage** section, see the file `docs/source/tracking.rst`. Mainly to inform users that if the LibYAML is not installed and they experience some performance issues, they can easily install it.

## How is this patch tested?

- tested it on macOS
- tested in Docker - Debian with and without LibYAML

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Improves performance of reading experiment data from local filesystem when LibYAML is installed

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [x] Docs
- [x] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Model Registry
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by pogil — CHANGES_REQUESTED** [review]

File: mlflow/utils/file_utils.py [hunk 1]
```diff
@@ -137,7 +141,7 @@ def write_yaml(root, file_name, data, overwrite=False):
 
     try:
         with codecs.open(yaml_file_name, mode='w', encoding=ENCODING) as yaml_file:
-            yaml.safe_dump(data, yaml_file, default_flow_style=False, allow_unicode=True)
+            yaml.dump(data, yaml_file, default_flow_style=False, allow_unicode=True, Dumper=SafeDumper)
```
[the remark this answers is not in the document]
Comment (Higgcz) [reply mlflow/utils/file_utils.py:144]: That’s a good question, let me check. I think I was just assuming it doesn’t use LibYAML by default, even if it’s available, but actually it makes more sense if it does.

[the remark this answers is not in the document]
Comment (Higgcz) [reply mlflow/utils/file_utils.py:144]: I've checked and unfortunately, those changes are necessary. PyYAML will install LibYAML bindings if LibYAML is available, but you still need to explicitly use `CLoader` and `CDumper`.

**Comment by Higgcz** [comment]
I've updated the code to satisfy linter, unfortunately we need to keep these changes in `file_utils.py`.

**Review by pogil — APPROVED** [review]
Thanks, looks good to me!

File: mlflow/utils/file_utils.py [hunk 2]
```diff
@@ -13,6 +13,10 @@
 from six.moves import urllib
 
 import yaml
+try:
+    from yaml import CSafeLoader as SafeLoader, CSafeDumper as SafeDumper
+except ImportError:
+    from yaml import SafeLoader, SafeDumper
```
Comment (mparkhe) [inline mlflow/utils/file_utils.py:19]: Can you use a different alias? Eg.

```
try:
    from yaml import CSafeLoader as YamlSafeLoader, CSafeDumper as YamlSafeDumper
except ImportError:
    from yaml import SafeLoader as YamlSafeLoader, SafeDumper as YamlSafeDumper
```

and use `YamlSafeLoader` and `YamlSafeDumper` in load and dump method.

Comment (Higgcz) [reply mlflow/utils/file_utils.py]: Done. One more question, should I squash the commits?

Comment (mparkhe) [reply mlflow/utils/file_utils.py]: No worries. I can squash commits when merging.

**Pull request merged**